code-challenges.test.js

```diff
@@ -15,42 +16,137 @@
+
+// ReferenceError: findLetter is not defined
 
 
 // b) Create the function that makes the test pass.
 
+// Create a function named findLetter that takes two inputs as a parameter: first one array and second parameter: single letter (string)
+// Create an empty array named filArray
+// Iterate over each value in the input array to find if the input letter/string is present
+    // If present, push the value into finArray
+// Return finArray
 
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
+const findLetter = (array,letter) => {
+    let filArray = []
+   array.map(value => {
+    if (value)array.includes(letter)
+    filArray = filArray.push()
+   })
+   return filArray
+    }
+
+// I also spent a bunch of time on this one and can't figure out what I'm doing wrong. I tried moving the return between different brackets and I tried removing the "if" and putting value into the .push. But now when I run it, it says this:
+
+// TypeError: filArray.push is not a function
+
+// 84 |    array.map(value => {
+// 85 |     if (value)array.includes(letter)
+// > 86 |     filArray = filArray.push()
+//    |                         ^
+// 87 |    })
+// 88 |    return filArray
+// 89 |     }
 
-// a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
+
+
+
+
+// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
+
+describe("fullHouse", () => {
+    const hand1 = [5, 5, 5, 3, 3]
 // Expected output: true
 const hand2 = [5, 5, 3, 3, 4]
 // Expected output: false
 const hand3 = [5, 5, 5, 5, 4]
 // Expected output: false
+    it("a function that takes in an array of 5 numbers and determines whether or not the array is a full house", () => {
+        expect(fullHouse(hand1)).toEqual(true)
+        expect(fullHouse(hand2)).toEqual(false)
+        expect(fullHouse(hand3)).toEqual(false)
+    })
+})
+
+
+// ● fullHouse › a function that takes in an array of 5 numbers and determines whether or not the array is a full house
+
+// ReferenceError: fullHouse is not defined
+
+// a) Create a test with expect statements using the variable provided.
+
+
+// Create a function named fullHouse that takes an array with 5 numbers in it as a parameter
+// Iterate over all the values in the array to determine if any of the indices values are equal. 
+// If there are equal values, determine the number of occurences of each value that was repeated
+// If the number of occurences of value1 or value 2 are equal to 3 AND the number of occurences of value1 or value2 are 2,
+    // Return "FULL HOUSE"
+    // Else return "No full house this time!"
 
 
 // b) Create the function that makes the test pass.
+
```

Comment:
I can undersrtand how it appears overwhelming. Did you start with the small pieces? Writing out in plain words what each step means to you. The pseudocode looks great. 

// Create a function named fullHouse that takes an array with 5 numbers in it as a parameter
`const fullHouse = (array) => { return array}`
// Iterate over all the values in the array to determine if any of the indices values are equal. 
`How do you reference values in an array? What operators do we use to get boolean values? Which one will be useful here?`
// If there are equal values, determine the number of occurences of each value that was repeated
`What can you use to count the occurrences? Or is the comparisons that you created previously enough?`
// If the number of occurences of value1 or value 2 are equal to 3 AND the number of occurences of value1 or value2 are 2,
`What tool can you use to show these conditions?`
    // Return "FULL HOUSE"
    // Else return "No full house this time!"